c2645637.lua

```diff
@@ -30,18 +30,21 @@ function c2645637.initial_effect(c)
 	c:RegisterEffect(e3)
 end
 function c2645637.disop(e,tp,eg,ep,ev,re,r,rp)
-	if Duel.GetChainInfo(ev,CHAININFO_TRIGGERING_LOCATION)==LOCATION_REMOVED and re:IsActiveType(TYPE_MONSTER) and rp==1-tp then
+	if rp==tp or e:GetHandler():IsStatus(STATUS_BATTLE_DESTROYED) then return end
+	if Duel.GetChainInfo(ev,CHAININFO_TRIGGERING_LOCATION)==LOCATION_REMOVED and re:IsActiveType(TYPE_MONSTER) then
 		Duel.NegateEffect(ev)
 	end
 end
 function c2645637.atkcon(e,tp,eg,ep,ev,re,r,rp)
 	return re:IsActiveType(TYPE_MONSTER) and Duel.GetChainInfo(ev,CHAININFO_TRIGGERING_LOCATION)==LOCATION_GRAVE
 end
-function c2645637.spfilter(c)
+function c2645637.spfilter(c,re,ev)
 	return c:GetSummonLocation()==LOCATION_GRAVE
+		and not (re and re:IsActivated() and re:IsActiveType(TYPE_MONSTER)
+		and Duel.GetChainInfo(ev,CHAININFO_TRIGGERING_LOCATION)==LOCATION_GRAVE)
```

Comment:
should be in `atkcon2` directly but not in filter

in fact, the most natural way is check event via `EFFECT_TYPE_CONTINUOUS` and activate the effect in `EVENT_CHAIN_END`

Reply:
it's my mistake, but i think EVENT_CHAINING is ok.